modules/service/src/main/scala/lucuma/odb/graphql/mapping/GmosDynamicMapping.scala

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.graphql
+package mapping
+
+import edu.gemini.grackle.TypeRef
+import io.circe.Json
+import io.circe.syntax.*
+import lucuma.core.enums.GmosAmpCount
+import lucuma.core.enums.GmosAmpGain
+import lucuma.core.enums.GmosAmpReadMode
+import lucuma.core.enums.GmosXBinning
+import lucuma.core.enums.GmosYBinning
+
+import table.GmosDynamicTables
+
+trait GmosDynamicMapping[F[_]] extends GmosDynamicTables[F] {
+
+  private def dynamicMapping[G, L, U](
+    typeRef: TypeRef,
+    table:   GmosDynamicTable[G, L, U]
+  ): ObjectMapping =
+    ObjectMapping(
+      tpe = typeRef,
+      fieldMappings = List(
+        SqlField("id",       table.Id, key = true),
+        SqlObject("exposure"),
+
+        SqlField("xBin",        table.CcdMode.Xbin,        hidden = true),
+        SqlField("yBin",        table.CcdMode.Ybin,        hidden = true),
+        SqlField("ampCount",    table.CcdMode.AmpCount,    hidden = true),
+        SqlField("ampGain",     table.CcdMode.AmpGain,     hidden = true),
+        SqlField("ampReadMode", table.CcdMode.AmpReadMode, hidden = true),
+
+        CursorFieldJson(
+          "readout",
+          cursor =>
+            for {
+              x <- cursor.field("xBin",        None).flatMap(_.as[GmosXBinning])
+              y <- cursor.field("yBin",        None).flatMap(_.as[GmosYBinning])
+              c <- cursor.field("ampCount",    None).flatMap(_.as[GmosAmpCount])
+              g <- cursor.field("ampGain",     None).flatMap(_.as[GmosAmpGain])
+              r <- cursor.field("ampReadMode", None).flatMap(_.as[GmosAmpReadMode])
+            } yield Json.obj(
+              "xBin"        -> x.asJson,
+              "yBin"        -> y.asJson,
+              "ampCount"    -> c.asJson,
+              "ampGain"     -> g.asJson,
+              "ampReadMode" -> r.asJson
+            ),
+            List("xBin", "yBin", "ampCount", "ampGain", "ampReadMode")
+        ),
```

Comment:
I resorted to a `CursorFieldJson` out of ignorance.  I wasn't sure how to make a single `GmosCcdMode` mapping that is the same for both GMOS North and South when two different tables would hold the fields.  Is there another way?

Reply:
Elsewhere we use `SqlObject` and then have a `SwitchMapping` that chooses the backing columns. Does that not work here?

Reply:
Yes, I had some mental block about it but that works.  I tried the same trick for the `GmosCustomMask` (which is currently implemented with a `CursorFieldJson`) and got a dreaded `AssertionError` from Grackle so I reverted that part and will open an issue.  At least until we execute sequences with a custom mask it shouldn't be an issue.